futures-util/src/async_await/join_mod.rs

```diff
@@ -0,0 +1,85 @@
+//! The `join` macro.
+
+use proc_macro_hack::proc_macro_hack;
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! document_join_macro {
+    ($join:item $try_join:item) => {
+        /// Polls multiple futures simultaneously, returning a tuple
+        /// of all results once complete.
+        ///
+        /// While `join!(a, b)` is similar to `(a.await, b.await)`,
+        /// `join!` polls both futures concurrently and therefore is more efficent.
+        ///
+        /// This macro is only usable inside of async functions, closures, and blocks.
+        /// It is also gated behind the `async-await` feature of this library, which is
```

Comment:
```suggestion
        /// It is also gated behind the `join-macro` feature of this library, which is
```

Reply:
Oh, it depends on what "this library" is referring to...

Reply:
Yeah. So, I would like to remove these descriptions and change them to use `doc(cfg(feature = ".."))` instead in the near future.